nixos/modules/services/x11/desktop-managers/plasma6.nix

```diff
@@ -0,0 +1,263 @@
+    # but kpackage is a library so we can't just wrap the one thing invoking it and be done.
+    # This also means things won't work for people not on Plasma, but at least this way it
+    # works for SOME people.
+    environment.sessionVariables.KPACKAGE_DEP_RESOLVERS_PATH = "${kdePackages.frameworkintegration.out}/libexec/kf6/kpackagehandlers";
+
+    # Enable GTK applications to load SVG icons
+    services.xserver.gdk-pixbuf.modulePackages = [pkgs.librsvg];
+
+    fonts.packages = [cfg.notoPackage pkgs.hack-font];
+    fonts.fontconfig.defaultFonts = {
+      monospace = ["Hack" "Noto Sans Mono"];
+      sansSerif = ["Noto Sans"];
+      serif = ["Noto Serif"];
+    };
+
+    programs.ssh.askPassword = mkDefault "${kdePackages.ksshaskpass.out}/bin/ksshaskpass";
+
+    # Enable helpful DBus services.
+    services.accounts-daemon.enable = true;
+    # when changing an account picture the accounts-daemon reads a temporary file containing the image which systemsettings5 may place under /tmp
+    systemd.services.accounts-daemon.serviceConfig.PrivateTmp = false;
+
+    services.power-profiles-daemon.enable = mkDefault true;
+    services.system-config-printer.enable = mkIf config.services.printing.enable (mkDefault true);
+    services.udisks2.enable = true;
+    services.upower.enable = config.powerManagement.enable;
+    services.xserver.libinput.enable = mkDefault true;
+
+    # Extra UDEV rules used by Solid
+    services.udev.packages = [
+      # libmtp has "bin", "dev", "out" outputs. UDEV rules file is in "out".
+      pkgs.libmtp.out
+      pkgs.media-player-info
+    ];
+
+    # Set up Dr. Konqi as crash handler
+    systemd.packages = [kdePackages.drkonqi];
+    systemd.services."drkonqi-coredump-processor@".wantedBy = ["systemd-coredump@.service"];
+
+    xdg.portal.enable = true;
+    xdg.portal.extraPortals = [kdePackages.xdg-desktop-portal-kde];
+    xdg.portal.configPackages = mkDefault [kdePackages.xdg-desktop-portal-kde];
+    services.pipewire.enable = mkDefault true;
+
+    services.xserver.displayManager = {
+      sessionPackages = [kdePackages.plasma-workspace];
+      defaultSession = mkDefault "plasma";
+    };
+    services.xserver.displayManager.sddm = {
+      # FIXME: extremely hacky wrapper
+      package = kdePackages.sddm.overrideAttrs (old: {
+        buildInputs = old.buildInputs ++ (with kdePackages; [kirigami qtsvg ksvg plasma5support qt5compat breeze-icons]);
+      });
+      theme = mkDefault "breeze";
+    };
+
+    security.pam.services = {
+      # FIXME: needs to use Qt6 kwallet somehow
+      # login.enableKwallet = true;
+      # kde.enableKwallet = true;
+      kde = {};
+      kde-fingerprint = lib.mkIf config.services.fprintd.enable { fprintAuth = true; };
+      kde-smartcard = lib.mkIf config.security.pam.p11.enable { p11Auth = true; };
+    };
+
+    programs.dconf.enable = true;
+    programs.firefox.nativeMessagingHosts.packages = [kdePackages.plasma-browser-integration];
```

Comment:
Can we finally bring kdeconnect to chromium https://github.com/NixOS/nixpkgs/pull/232528 ?

Reply:
You mean p-b-i? I'm not sure, I haven't tested it, but if your PR works, you can add that.

Reply:
Yes, exactly.

Can we merge #232528 beforehand?

Then we only need to set the following and things work:

```
programs.chromium = {
  enablePlasmaBrowserIntegration = true;
  plasmaBrowserIntegrationPackage = pkgs.kdePackages.plasma-browser-integration;
};
```

![image](https://github.com/NixOS/nixpkgs/assets/7258858/f9174c09-7334-4670-9e23-9067af627a49)

Reply:
I mean, I don't use Chromium, so I'm not the one to make that call, but it looks OK.

Reply:
You can find and switch to the Tabs from KRunner, so I call it works :)

Reply:
The linked PR got merged. Can you add the following to enable the browser integration?

```
  programs.chromium = {
    enablePlasmaBrowserIntegration = true;
    plasmaBrowserIntegrationPackage = pkgs.kdePackages.plasma-browser-integration;
  };
```

Reply:
Done.